ddl/column.go

```diff
@@ -133,6 +133,10 @@ func (d *ddl) onAddColumn(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
+	if err = checkAddColumnTooManyColumns(len(tblInfo.Columns)); err != nil {
```

Comment:
Why should we change this? From the PR's title, I guess it only changes the tests.

Reply:
@shenli this is to modify this PR#6778.

Reply:
I think `len(tblInfo.Columns) +1` doesn't always right in here. You need to put this check to line 158.
And why this PR passes tests? Do we need to update test?